[taheryasir/step 10] Sort the task list in order of creation date and time

taheryasir opened the pull request:
- Currently, they are sorted in order of ID, but let's sort them in descending order of creation date and time.
- Let's write in the system spec that the sorting is working well

yo-it-engineer commented on myapp/spec/controllers/tasks_controller_spec.rb:
```diff
@@ -1,6 +1,9 @@
 require 'rails_helper'
+require 'pp'
```
Seemed that we didn't need to require 'pp' as user.
ref: https://qiita.com/scivola/items/b4851f3ebe43a0dba9c5

By the way, where do you use this inside this file?

taheryasir replied:
i added this for debugging but i guess it's not needed after debugging, will remove this

yo-it-engineer commented on myapp/spec/controllers/tasks_controller_spec.rb:
```diff
@@ -48,4 +51,19 @@
+    task1_index = res_str.index('task1')
+    task2_index = res_str.index('task2')
+
+    expect(task1_index).to be < task2_index
```
wow. I didn't know you can do comparison like that. 

Well it work. 
But I'm sorry to tell you that the english instruction got wrong translation on it.

It was suppose to instruct you to write "feature spec" for this step.
https://relishapp.com/rspec/rspec-rails/docs/feature-specs/feature-spec

English | Japanese
----- | -----
![image](https://user-images.githubusercontent.com/10822260/122311504-4f039b80-cf4d-11eb-9d68-cf9d470b91ec.png) | ![image](https://user-images.githubusercontent.com/10822260/122311547-693d7980-cf4d-11eb-8525-e62bd3c77cde.png)

With capybara's API, it's easier to do the comparison. 
you can parse the text on the title field and compare it with the expected order of the task.

ref(if needed): 

https://github.com/Fablic/training/pull/405/files#diff-82689603be06b3081543d76c323b7476b917358f81297e164b83cdf1ebbe8359R66-R83

taheryasir replied:
Done

yo-it-engineer reviewed (commented):
It's easier for reviewer to check the wiki page of the training program if you include the link to that branch "develop_en".
I see you already using FactoryBot. That's great. We are also using that in the production code for specs.

Please change the spec part from controller test to "feature spec"(more instruction on comment). The rest seemed good.

taheryasir commented on myapp/app/controllers/tasks_controller.rb:
```diff
@@ -4,6 +4,7 @@ class TasksController < ApplicationController
   # GET /tasks or /tasks.json
   def index
     @tasks = Task.all
+    @tasks = Task.all.order(created_at: 'desc')
```
fixed

yo-it-engineer commented on myapp/app/views/tasks/_form.html.erb:
```diff
@@ -13,27 +13,28 @@
   <div class="field mb-3">
     <%#= form.label :label %>
-    <%= form.text_field :label, class: "form-control", placeholder: "Label" %>
+    <%= form.text_field :label, required: true, class: "form-control", placeholder: "Label" %>
```
[suggestion]
hmm. 
Since your model validator also didn't check, maybe we didn't need to this field to be required. (well that depend on how you want your apps to be like. I will leave that to your decision)

taheryasir replied:
actually i will add model validations in next step, so that's why added this.

yo-it-engineer commented on myapp/app/views/tasks/index.html.erb:
```diff
@@ -8,21 +8,23 @@
-        <td><%= task.name %></td>
+        <td class="task-name"><%= task.name %></td>
         <td><%= task.desc %></td>
         <td><%= task.status %></td>
```
since you have the enum setup on the "create task" field, let's update here to show the status(and priority) with the text from enum!

![image](https://user-images.githubusercontent.com/10822260/122487143-863d7f80-d015-11eb-9643-13926beef88d.png)
![image](https://user-images.githubusercontent.com/10822260/122487154-8d648d80-d015-11eb-93ef-2ddb04e6a1b0.png)

example: 
`Task.statuses.key(task.status)`

source: 
https://stackoverflow.com/a/45374287

yo-it-engineer reviewed (commented):
Just a few more comment. 
Glad to see the system spec is running now.

Oh ya, when you done fixing and want the reviewer to review again, please do so by clicking on the 'Re-request review' button. So that the reviewer know that your changes is ready for review.

![image](https://user-images.githubusercontent.com/10822260/122488265-12e93d00-d018-11eb-917a-4d43aab38a56.png)

yo-it-engineer commented:
btw, one more question. How do you run rspec on your environment? I tried to run with my environment with terminal but it failed:

![image](https://user-images.githubusercontent.com/10822260/122489797-6c9f3680-d01b-11eb-8bd8-5c20d79cffa3.png)

taheryasir commented on myapp/docker-compose.yml:
```diff
@@ -27,6 +30,11 @@ services:
       DB_PASSWORD: password
       DB_HOST: db
 
+  chrome:
+    image: selenium/standalone-chrome:latest
+    ports:
+      - 4444:4444
```
Ok i will add the reference in develop_en

taheryasir commented:
> btw, one more question. How do you run rspec on your environment? I tried to run with my environment with terminal but it failed:
> 
> ![image](https://user-images.githubusercontent.com/10822260/122489797-6c9f3680-d01b-11eb-8bd8-5c20d79cffa3.png)

for now i am posting here, since there will be other reviewers too in future i will add these in project readme

to run docker file `docker-compose up --build`
to run `rspec` you can run inside container or directly from app root
`docker exec -it myapp_api_1 bundle exec rspec`

myapp_api_1 - container name

yo-it-engineer commented:
> for now i am posting here, since there will be other reviewers too in future i will add these in project readme

> to run docker file docker-compose up --build
to run rspec you can run inside container or directly from app root
docker exec -it myapp_api_1 bundle exec rspec

> myapp_api_1 - container name

Thanks for the step. it seemed that my way of running is fine too. I'm just having library problem maybe.
[memo] commenting out this line worked for me:
![image](https://user-images.githubusercontent.com/10822260/122517292-cb2fd900-d04a-11eb-9360-a2a692f0b406.png)

yo-it-engineer reviewed (commented):
I checked the changes! Almost LGTM, but seemed that adding the required field caused some specs to fail. can you fix them?

I think you need to add in priority, status to factory/task.rb.

taheryasir commented:
removed 

> I checked the changes! Almost LGTM, but seemed that adding the required field caused some specs to fail. can you fix them?

added 
> I think you need to add in priority, status to factory/task.rb.

yo-it-engineer reviewed (approved):
LGTM! 
![Image from iOS (85)](https://user-images.githubusercontent.com/10822260/122519557-8eb1ac80-d04d-11eb-802f-603d3ff28748.jpg)
